### ISSUE TITLE
Duplicate transaction date

### ISSUE THREAD

**Issue by gabriele-v** [moneymanagerex/moneymanagerex#3784]
MMEX version:
 - [x] 1.5.5
 - [ ] 1.6.x

Operating System:
 - [x] Windows
 - [x] Mac OSX
 - [x] Linux 

# Description of the bug
When duplicating a transaction, the date is automatically set to today. Was working fine, not sure when it changed, probably in 1.5.x?

# Reproduction

Is the bug reproducible? 
 - [x] Always 
 - [ ] Randomly 
 - [ ] Happened only once

Reproduction steps:
1. Duplicate a transaction

Expected result:
- Transaction is duplicated exactly like the original one, all fields

Actual result:
- Date is set to today

# Additional information
The flag in options "use original date when copying/pasting transactions" works as expected, while duplication IMHo should always keep the original date, because it's a "duplication". In my use case, I leave the "use original date when copying/pasting transactions" disabled, so if I need to do a transaction with same date I use duplicate, otherwise I use copy paste.

**Comment by whalley** [comment]
https://github.com/moneymanagerex/moneymanagerex/issues/2982

The logic was changed based on this ticket. Perhaps we should use the "use original date when copying/pasting transactions" option to override this for duplication salso.

**Comment by Dandy64** [comment]
I am not sure it should be changed. All the versions I have been using have the same behavior - after duplication of a transaction the date is set to today. I suppose it is OK. 

If I take an **old** transaction and I want to duplicate it, it looks quite reasonable that I am going to use the same or very similar data like category, payee, free text, etc. If I do it today probably I am going to use the current day, perhaps yesterday or even a few days ago, but it is very unlikely I need the old date of the old transaction.

**Comment by gabriele-v** [comment]
For sure before #2982 duplicate was keeping the same date, while copy/paste didn't change and it was and it's managed by the options flag. 

I find useful to duplicate keeping the same date for example when I don't insert transactions for a while and I need for example to insert 3 transactions for the same date in the past, different only for the category.

**Comment by whalley** [comment]
New option added
![image](https://user-images.githubusercontent.com/487345/130337322-4ac6bcc0-9d69-497c-a4d7-32897cef0fba.png)

**Comment by DavidCinUK** [comment]
It thought I'd add to this thread as I use the "Create a Duplicate" option to create a number of "future" transactions so it's easier to see how those future transactions will effect an account's balance.
If the date of the "new duplicate" keeps the date of that future event it would be very easy to move the "new duplicate"'s future date one month forward ... whereas, at the moment, when that "new duplicate of a future event" gets created it takes today's date, so the whole date data needs to be changed to the "new duplicate"'s date.
